submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,352 @@
+        </div>
+      </fieldset>
+      
+      <fieldset class="filter filter-age">
+        <p>sort by age</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="age09" name="sort-age">
+        <label for="age09" class="radio-lable">young - old</label>
+        <input type="radio" id="age90" name="sort-age">
+        <label for="age90" class="radio-lable">old - young</label>
+        <input type="radio" id="ageAll" name="sort-age">
+        <label for="ageAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+
+      
+      <fieldset class="filter filter-search">
+        <p>Search user by email</p>
+        <input id="nameInput" type="text" size="30" class="searchEmail icon search">
+      </fieldset>
+    `;
+  }
+
+  renderFilters() {
+    this.filtersContainer.insertAdjacentHTML('beforeend', this.getHtmlFilter());
+  }
+}
+
+class Filter {
+  constructor(renderer) {
+    this.profilesContainer = document.querySelector('.profile-cards');
+    this.renderDataWithFilters = renderer;
+    renderer.renderFilters();
+    this.filtersPanel = document.querySelector('#filters');
+    this.searchInput = document.querySelector('#nameInput');
+    this.sortedUsers = [];
+    this.sortIndicators = {
+      age: 'age',
+      name: 'name',
+    };
+    this.sortMethods = {
+      byGender: 'sort-gender',
+      byAge: 'sort-age',
+      byName: 'sort-name',
+    };
+    this.sortByNameTypes = {
+      inAscending: 'nameAZ',
+      inDescending: 'nameZA',
+      byDefault: 'nameAll',
+    };
+    this.sortByAgeTypes = {
+      inAscending: 'age09',
+      inDescending: 'age90',
+      byDefault: 'ageAll',
+    };
+    this.sortByGenderTypes = {
+      male: 'male',
+      female: 'female',
+    };
+  }
+
+  uncheckRadioButton(node) {
+    const checkedRadio = document.querySelector(
+      `input[type=radio][name=${node.buttonName}]:checked `
+    );
+    if (checkedRadio) {
+      checkedRadio.checked = false;
+    }
+  }
+
+  selectSortMethod(selectedRadioButton) {
+    const methodName = selectedRadioButton.name;
+    const sortMethod = {
+      [this.sortMethods.byGender]: () => {
+        this.uncheckRadioButton({buttonName : this.sortMethods.byName});
+        this.uncheckRadioButton({buttonName : this.sortMethods.byAge});
+        this.sortByGender(selectedRadioButton.id);
+      },
+      [this.sortMethods.byName]: () => {
+        this.uncheckRadioButton({buttonName : this.sortMethods.byAge});
+        this.sortByContent(
+          selectedRadioButton.id,
+          this.sortByNameTypes,
+          this.sortIndicators.name
+        );
+      },
+      [this.sortMethods.byAge]: () => {
+        this.uncheckRadioButton({buttonName : this.sortMethods.byName});
+        this.sortByContent(
+          selectedRadioButton.id,
+          this.sortByAgeTypes,
+          this.sortIndicators.age
+        );
+      },
+    };
+    return sortMethod[methodName]();
+  }
+
+  startFilter() {
```

Comment:
You not starting filter process, you add listeners :)

Reply:
Ok, renamed this method to ```addListenersForFilters() ```